Add ai-telemetry skeleton folders and zookeeper

RH-csaggin opened the pull request:
- Added ai-telemetry skeleton folders 
- Zookeeper:
  - Moved to kustomization structure.
  - Added custom pvc (removed from the integrated template on sts).
  - Added `namePrefix: obs-` for a better identification.
- Tested on lab cluster with `oc kustomize overlays/nerc-ocp-obs | oc create -f - -n zookeeper-obs`

RH-csaggin committed:
adding zookeper deplyment for ai-telemetry on obs cluster

computate commented on ai-telemetry/base/zookeeper/statefulset.yaml:
```diff
@@ -0,0 +1,172 @@
+            defaultMode: 493
+        - name: data
+          persistentVolumeClaim:
+            claimName: zookeeper-data
```
@RH-csaggin I don't think we need this volume or the PersistentVolumeClaim below if we configure the `volumeClaimTemplates` instead: 
```yaml
  volumeClaimTemplates:
    - apiVersion: v1
      kind: PersistentVolumeClaim
      metadata:
        name: data
      spec:
        accessModes:
          - "ReadWriteOnce"
        resources:
          requests:
            storage: "10Gi"
```

computate commented on ai-telemetry/overlays/nerc-ocp-obs/kustomization.yaml:
```diff
@@ -0,0 +1,7 @@
+- ../../base
+- pvc
+
+namePrefix: obs-
```
Let's remove the `namePrefix` and the `pvc` here, I don't think we need it.

computate commented on ai-telemetry/base/zookeeper/kustomization.yaml:
```diff
@@ -0,0 +1,9 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Let's add the namespace here with 0 indent below `kind: Kustomization`: 
```yaml
namespace: zookeeper
```
This will apply the namespace to all resources that this includes.

computate commented on ai-telemetry/base/zookeeper/statefulset.yaml:
```diff
@@ -0,0 +1,172 @@
+            limits:
+              cpu: 375m
+              ephemeral-storage: 2Gi
+              memory: 384Mi
```
Let's give zookeeper `'1'` whole cpu, and `1Gi` memory limit here. This prevents the container CPU cycles from being rate limited which slows the application down.

computate commented on ai-telemetry/base/zookeeper/statefulset.yaml:
```diff
@@ -0,0 +1,172 @@
+    app.kubernetes.io/component: zookeeper
+    role: zookeeper
+spec:
+  replicas: 1
```
Let's try scaling the replicas here to 3 for our cluster to test out high availability.

RH-csaggin committed:
fixing computate comments

- added back the volumeClaimTemplates and removed pvc to allow scaling.
- prefix to have it more agnostic.
- add namespace to base/zookeeper/kustomization to apply to all the resoruces.
- rectify limits preventing slowing the application.
- set replica to 3 for production ready HA.
- created namespace under cluster scope to host the zookeeper service

schwesig reviewed (approved):
checked together this morning
looking good!

computate reviewed (changes requested):
Let's rerun the helm chart with `replicaCount=3` to fix some StatefulSet configuration for 3 replicas.

RH-csaggin committed:
Merge branch 'OCP-on-NERC:main' into main
